sbol_utilities/helper_functions.py

```diff
@@ -80,6 +80,18 @@ def toplevel_named(doc: sbol3.Document, name: str) -> Optional[sbol3.Identified]
         raise ValueError(f'Name is not unique: {name}')
 
 
+def filter_top_level(doc: sbol3.Document, filter: Callable[[sbol3.TopLevel], bool]) -> Iterable[sbol3.TopLevel]:
+    """Filters and returns iterable of TopLevel Objects in a document,
+    which match a criteria set by a callable argument.
+
+    :param doc: SBOL Document to search
+    :param filter: Callable acting as filter on List of TopLevel objects
+    :return: TopLevel iterator satisfying given filter
+    """
+    result: List[sbol3.TopLevel] = [obj for obj in doc.objects if filter(obj)]
```

Comment:
Rather than making a list and turning it into an iterable, you can return an iterable direcly by using a generator () instead of a list []

Reply:
Oh alright.